python/lsst/ts/MTAlignment/alignmentCSC.py

```diff
@@ -0,0 +1,198 @@
+        self.config = config
+
+    @staticmethod
+    async def get_config_pkg(self):
+        return "ts_config_mttcs"
+
+    async def do_measureTarget(self):
+        """Measure and return coordinates of a target. Options are
+        M1M3, M2, CAM, and DOME"""
+        pass
+
+    async def do_align(self):
+        """Perform correction loop"""
+        pass
+
+    async def do_healthCheck(self):
+        """run healthcheck script"""
+        pass
+
+    async def do_laserPower(self):
+        """put the laser in sleep state"""
+        pass
+
+    async def do_powerOff(self):
+        """full power off of tracker and interface"""
+        pass
+
+    async def do_measurePoint(self):
+        """measure and return coords of a specific point"""
+        pass
+
+    async def do_pointDelta(self):
+        """return a vector between two points"""
+        pass
+
+    async def do_setReferenceGroup(self):
+        """Nominal point group to locate tracker station to and provide data
+        relative to"""
+        pass
+
+    async def do_setWorkingFrame(self):
+        """attempt to set the passed string as the SpatialAnalyzer working
+        frame"""
+        pass
+
+    async def do_halt(self):
+        """halts any executing measurement plan and returns to ready state"""
+        pass
+
+    async def do_loadSATemplateFile(self):
+        """SA Template file path and name. This is in the filesystem on the
+        T2SA host."""
+        pass
+
+    async def do_measureDrift(self):
+        """measure tracker drift"""
+        pass
+
+    async def do_resetT2SA(self):
+        """reboots t2sa and SA"""
+        pass
+
+    async def do_newStation(self):
+        """create new tracker station"""
+        pass
+
+    async def do_saveJobfile(self):
+        """save job file"""
+        pass
+
+    async def correction_loop(self):
+        await self.model.set_telescope_position(
+            self.elevation, self.azimuth, self.camrot
+        )
+
+        await self.model.measure_m1m3()
+        aligned = False
+        loopcount = 1
+        while not aligned:
+            print("Loop Number: " + str(loopcount))
+            await self.model.measure_cam()
+            await self.model.measure_m2()
+
+            camOffset = self.parse_offsets(await self.model.query_cam_offset("CAM"))
+            m2Offset = self.parse_offsets(await self.model.query_m2_offset("M2"))
+
+            if self.in_tolerance(camOffset) and self.in_tolerance(m2Offset):
+                break
+            elif loopcount >= self.max_iters:
+                break
+            else:
+                loopcount += 1
+
+    def parse_offsets(self, t2sa_string):
+        """
+        Takes a string containing spatial coordinates  from T2SA, and returns
+        a dict with the following keys: RefFrame, X, Y, Z, Rx, Ry, Rz, and
+        Timestamp
+
+        Parameters
+        ----------
+
+        t2sa_string : `str`
+            the ascii string from T2SA
```

Comment:
Please describe the format of the string. Also, what happens if the string doesn't meet the expected format? Some error handling code would be welcome.